### DISCUSSION TITLE
EMP does 100x as much damage as expected to junctions

### DISCUSSION THREAD

**Discussion by BlackMoons** [FakeFishGames/Barotrauma#14481] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

While trying out various explosion commands, I found the following:

EMP damage does 100x as much condition damage as any other type of damage. 1 EMP damage is enough to remove 95 to 99% of the condition of junction boxes. 0.5 EMP damage removes 50% of their condition.

ie, explosion 10000 0 0 0 0 1

In comparison to regular damage, it takes 100 damage to destroy a junction box, ie explosion 10000 0 100, or 50 to remove half of the health. (Oddly enough even 10 damage, ie explosion 10000 0 10 also KO'ed every single character on the sub, meaning players are extremely vulnerable to explosion damage, im guessing it hit every limb and was added up, with no limit on the number of limbs it hit)

I think this is must be a bug due to the fact EMP grenades do 100 EMP damage, 100x as much needed to destroy any electrical device, and nuclear railgun shell has 50 EMP damage. 

### Reproduction steps

Load any sub.
into console type: "explosion 10000 0 0 0 0 1"

Watch as all your junctions are at 1~5% health. 

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Ek-Videogames** [comment]
Oh... EMP integers accidentally acting like percentage of 'max health'?

[I knew that sounded familiar.](https://github.com/FakeFishGames/Barotrauma/issues/5434#issue-849777982)

**Reply by BlackMoons** [reply]
Yea, I feel the solution here would be make EMP damage either be percent, as in 0~100 percent of item condition, if for some reason you wanted to keep EMP damage doing a percent of health, or make it flat damage against max health, since then it would fulfill the standard expectation that 1 = little damage, and 100 = lots of damage that every other damage type has. Id be OK with either since all(?) vanilla EMPable devices only have 100 max health anyway, they would work more or less the same. 

Nobody seems to remember that EMP damage is a 0 to 1 scale instead of 0 to 100 scale when setting their weapons up. Lots of vanilla and modded items currently do EMP damage in the 30 to 100 range, iv not seen anything setup that assumes its a 0 to 1 scale.

**Comment by Regalis11** [comment]
This is not actually a bug: the strength of the EMP is configured in the range of 0-1 - it doesn't represent the amount of damage, but the "strength of the pulse". While I agree this is inconsistent with other types of damage, I feel it'd be very risky to change this at this point, because it'd cause EMPs to suddenly become a hundred times less effective in all mods that use EMPs. But we should definitely take a look at vanilla items (many of which seem to assume the value is in the range of 0-100) and somehow make it clearer it's between 0-1.

**Reply by BlackMoons** [reply]
Iv yet to see any mods that actually got it right that its 0-1, they all seem to assume its 0-100 too.

All vanilla items seem to assume the range of 0-100 from what I can see, I feel its much safer to make the damage fit everyone assumption then to try and fix every single vanilla item and notify the mod makers that this one type of damage operates different then every other type.

Edit: Seems I may have been confusing the EMP affliction with EMP strength from explosions. Vanilla items do seem to have a reasonable empstrength value. Sorry my bad.
